From e070af31638acf77ac651cc56a5fd98eecf4d572 Mon Sep 17 00:00:00 2001
From: Ilya Fedin <fedin-ilja2010@ya.ru>
Date: Sun, 20 Jul 2025 14:24:32 +0000
Subject: [PATCH] Limit malloc_trim to once per second

---
 .../platform/linux/integration_linux.cpp            | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/Telegram/SourceFiles/platform/linux/integration_linux.cpp b/Telegram/SourceFiles/platform/linux/integration_linux.cpp
index f36bfb632630ec..1a5e3b6903df54 100644
--- a/Telegram/SourceFiles/platform/linux/integration_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/integration_linux.cpp
@@ -176,7 +176,18 @@ LinuxIntegration::LinuxIntegration()
 	QObject::connect(
 		QCoreApplication::eventDispatcher(),
 		&QAbstractEventDispatcher::aboutToBlock,
-		[] { malloc_trim(0); });
+		[] {
+			static auto timer = [] {
+				QElapsedTimer timer;
+				timer.start();
+				return timer;
+			}();
+
+			if (timer.hasExpired(1000)) {
+				malloc_trim(0);
+				timer.start();
+			}
+		});
 #endif // __GLIBC__
 }
 
